controllers/glanceapi_controller.go

```diff
@@ -1700,3 +1756,46 @@ func (r *GlanceAPIReconciler) GetHorizonEndpoint(
 
 	return ep, nil
 }
+
+// verifyApplicationCredentials checks if ApplicationCredential secret exists and adds it to configVars
+// The AC secret is created by the keystone-operator's AC controller when the AC is ready.
+// If the secret exists and is valid, we use AC auth. Otherwise, we fall back to password auth.
+func (r *GlanceAPIReconciler) verifyApplicationCredentials(
```

Comment:
This functions seems very generic to me: if we do not pass `instance *glancev1.GlanceAPI`, but we simply pass a `namespace` or a `serviceName` there would be nothing Glance specific here, and the service could consume it from keystone-operator.
In addition, this might result in a lot of code duplication across the board, so I would invest some time to make it generic and remove the `instance` dependency.